src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -95,73 +94,178 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+      if (request.comments === null) {
+        request.comments = '';
+      }
+      studentsEnrollRequest.studentEnrollRequests.push(request);
+    }
+
+    // Team name should be unique among all sections of a course.
+    const teamInSection: Map<string, string> = new Map();
+    for (const student of this.existingStudents) {
+      teamInSection.set(student.teamName, student.sectionName);
+    }
+    for (const request of studentsEnrollRequest.studentEnrollRequests) {
+      if (teamInSection.has(request.team) && teamInSection.get(request.team) !== request.section) {
+        this.statusMessageService.showErrorMessage(
+            `Team "${request.team}" is detected in both Section "${request.section}"
+             and Section "${teamInSection.get(request.team)}".
+             Please use different team names in different sections`);
+        return;
+      }
+
+      teamInSection.set(request.team, request.section);
+    }
+
+    this.studentService.enrollStudents(this.courseid, studentsEnrollRequest).subscribe((resp: Students) => {
+      const enrolledStudents: Student[] = resp.students;
+      this.showEnrollResults = true;
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
+      this.enrollResultPanelList =
+          this.propagateEnrollResultPanelList(this.existingStudents, enrolledStudents,
+              studentsEnrollRequest.studentEnrollRequests);
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showErrorMessage(resp.error.message);
+    });
+    this.studentService.getStudentsFromCourse(this.courseid).subscribe((resp: Students) => {
+      this.existingStudents = resp.students;
+    });
+  }
+
+  private propagateEnrollResultPanelList(existingStudents: Student[], enrolledStudents: Student[],
+                                         enrollRequests: StudentEnrollRequest[]): EnrollResultPanel[] {
+    const updateStatus: { [key: string]: number } = {
+      new: 0,
+      modified: 1,
+      modifiedUnchanged: 2,
+      error: 3,
+      unmodified: 4,
     };
-    this.httpRequestService.put('/students', paramMap, this.enrollData)
-        .subscribe((resp: EnrollResultPanelList) => {
-          this.showEnrollResults = true;
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
-          this.enrollResultPanelList = resp.enrollResultPanelList;
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessage.pop(); // removes any existing error status message
-          this.statusMessage.push({
-            message: resp.error.message,
-            color: 'danger',
-          });
+    const panelClass: { [key: number]: string } = {
+      0: 'bg-primary',
```

Comment:
Expose the enum to the HTML file and put the logic of choosing different CSS class in the HTML file.